File: backend_old/mysql/table/performance.go
package table

import (
	"time"

)

type Performance struct {
	PerformanceId int64 `json:"PerformanceId"`
	UserId int64 `json:"UserId"`
	Year int `json:"Year"`
	Month int `json:"Month"`
	BanchId int64 `json:"BanchId"`
	Goal string `json:"Goal"`
	Attitude int `json:"Attitude"`
	Efficiency int `json:"Efficiency"`
	Professional int `json:"Professional"`
	Directions string `json:"Directions"`
	BeLate int `json:"BeLate"`
	DayOffNotOnRule int `json:"DayOffNotOnRule"`
	BanchName string `json:"BanchName"`
	CreateTime time.Time `json:"CreateTime"`//創建的時間
	LastModify time.Time `json:"LastModify"`// 上次修改的時間
}

type PerformanceExtend struct {
	Performance
	UserName string `json:"UserName"` // 名字
	CompanyId int64 `json:"CompanyId"` // 公司編號
}

type YearPerformance struct {
	UserId int64
	Year int
	UserName string
	Avg float32
}

File: frontend_web/src/types/performance.ts
import companyBanchTypes from './companyBanch'
import userTypes from './user'

declare namespace performanceTypes {
    interface TABLE {
        CompanyId: number
        UserId: number
        PerformanceId: number
        Year: number
        Month: number
        BanchId: number
        Goal: string
        Attitude: number
        Efficiency: number
        Professional: number
        Directions: string
        BeLate: number
        DayOffNotOnRule: number
        DeleteFlag: Flag
        DeleteTime: string
        CreateTime: string
        LastModify: string
    }

    interface year {
        UserName: userTypes.TABLE['UserName']
        Year: TABLE['Year']
        Score: number
    }

    interface reducerType {
        all: Array<
        TABLE & {
            UserName: userTypes.TABLE['UserName']
            RoleId: userTypes.TABLE['RoleId']
            BanchName: companyBanchTypes.TABLE['BanchName']
        }>
        year: year[]
    }
}
export default performanceTypes
